src/EnergyPlus/Plant/PlantManager.cc

```diff
@@ -1314,8 +1315,12 @@ void GetPlantInput(EnergyPlusData &state)
                         // now deal with demand components of the ZoneHVAC type served by ControlCompOutput
                         break;
                     }
+                    case PlantEquipmentType::SwimmingPool_Indoor: {
+                        this_comp.CurOpSchemeType = OpScheme::Demand;
+                        this_comp.compPtr = SwimmingPool::SwimmingPoolData::factory(state, CompNames(CompNum));
+                        break;
+                    }
```

Comment:
This looks exactly right.  I think there should be one more spot in PlantManager where we need to switch on equipment type.  Down where we assign the flow request priority.  If that's the only change needed here, I can add that while I'm doing final testing.